samples/snippets/standard/assembly/unloading/unloadability_issues_example_main.cs

```diff
@@ -45,11 +46,14 @@ static int ExecuteAndUnload(string assemblyPath, out WeakReference testAlcWeakRe
 
             var args = new object[1] {new string[] {"Hello"}};
 
-            // Issue preventing unloading #1 - we keep MethodInfo of a method for an assembly loaded into the TestAssemblyLoadContext in a static variable
+            // Issue preventing unloading #1 - we keep MethodInfo of a method
+            // for an assembly loaded into the TestAssemblyLoadContext in a static variable.
             entryPoint = new TestInfo(a.EntryPoint);
             testEntryPoint = a.EntryPoint;
 
-            int result = (int)a.EntryPoint.Invoke(null, args);
+            var oResult = a.EntryPoint?.Invoke(null, args);
+            int result = (oResult is not null) ? (int)oResult : -1;
+
             alc.Unload();
 
             return result;
```

Comment:
I might rework this a bit:

```suggestion
            alc.Unload();
            return (oResult is int result) ? result : -1;
```